### PR TITLE
Add dynamic fee teal test

### PR DESCRIPTION
**Pull request by justicz**
Added a test for the dynamic fee teal script

### CODE REVIEW

File: test/scripts/dynamic-fee-teal-test.sh [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+trap "goal network stop -r ${NETDIR}; rm -rf ${TEMPDIR}" 0
+
+export ALGORAND_DATA=${NETDIR}/Node
+
+ACCOUNT=$(goal account list|awk '{ print $3 }')
+ACCOUNTB=$(goal account new|awk '{ print $6 }')
+ACCOUNTC=$(goal account new|awk '{ print $6 }')
+ACCOUNTD=$(goal account new|awk '{ print $6 }')
+ZERO_ADDRESS=AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAY5HFKQ
+LEASE=YmxhaCBibGFoIGxlYXNlIHdoYXRldmVyIGJsYWghISE=
```
Comment (tsachiherman) [inline test/scripts/dynamic-fee-teal-test.sh]: is that a base64 encoding ? if so, can you also try to add "\" to break it ?

File: test/scripts/dynamic-fee-teal-test.sh [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+goal network create -r ${NETDIR} -n tbd -t ${GOPATH}/src/github.com/algorand/go-algorand/test/testdata/nettemplates/TwoNodes50EachFuture.json
+
+goal network start -r ${NETDIR}
+
+# replaces prior trap0
+trap "goal network stop -r ${NETDIR}; rm -rf ${TEMPDIR}" 0
+
+export ALGORAND_DATA=${NETDIR}/Node
+
+ACCOUNT=$(goal account list|awk '{ print $3 }')
```
Comment (tsachiherman) [inline test/scripts/dynamic-fee-teal-test.sh]: you're using pipe here, which is ok, but you need to ensure that you propogate the error code from the goal and not overriding it with the exit code from awk. use "set -o pipefail" to do that.

File: test/scripts/dynamic-fee-teal-test.sh [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+TEMPDIR=$(mktemp -d)
+trap "rm -rf $TEMPDIR" 0
+
+NETDIR=${TEMPDIR}/net
+
+if [ ! -z $BINDIR ]; then
+    export PATH=${BINDIR}:${PATH}
+fi
+
+goal network create -r ${NETDIR} -n tbd -t ${GOPATH}/src/github.com/algorand/go-algorand/test/testdata/nettemplates/TwoNodes50EachFuture.json
```
Comment (tsachiherman) [inline test/scripts/dynamic-fee-teal-test.sh]: nit: some more meaningful network names would be nice.

File: test/scripts/dynamic-fee-teal-test.sh (hunk 1 above)
Comment (justicz) [reply test/scripts/dynamic-fee-teal-test.sh]: done

File: test/scripts/dynamic-fee-teal-test.sh (hunk 2 above)
Comment (justicz) [reply test/scripts/dynamic-fee-teal-test.sh]: done

File: test/scripts/dynamic-fee-teal-test.sh (hunk 3 above)
Comment (justicz) [reply test/scripts/dynamic-fee-teal-test.sh]: done

File: test/scripts/dynamic-fee-teal-test.sh [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+TEMPDIR=$(mktemp -d)
+trap "rm -rf $TEMPDIR" 0
+
+NETDIR=${TEMPDIR}/net
+
+if [ ! -z $BINDIR ]; then
+    export PATH=${BINDIR}:${PATH}
+fi
+
+goal network create -r ${NETDIR} -n dynamic-fee-teal-test -t ${GOPATH}/src/github.com/algorand/go-algorand/test/testdata/nettemplates/TwoNodes50EachFuture.json
```
Comment (pzbitskiy) [inline test/scripts/dynamic-fee-teal-test.sh]: might not use TwoNodes50EachFuture anymore, TwoNodes50Each also works

File: test/scripts/dynamic-fee-teal-test.sh [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+algotmpl -d ${GOPATH}/src/github.com/algorand/go-algorand/tools/teal/templates/ dynamic-fee --amt=1000000 --cls=${ACCOUNTC} --to=${ACCOUNTD} --fv=1 --lv=1001 --lease=${LEASE} > ${TEMPDIR}/dynamic.teal
+
+#
+# Fee will come from ACCOUNT in the first transaction
+# We will close out to ACCOUNTC in second transaction
+# We will pay ACCOUNTD with 1000000 microAlgos in second transaction
+#
+
+# Compile the template
+goal clerk compile -a ${ACCOUNTB} -s ${TEMPDIR}/dynamic.teal -o ${TEMPDIR}/dynamic.steal
```
Comment (pzbitskiy) [inline test/scripts/dynamic-fee-teal-test.sh]: steal is used for Derek's lang if I remember correctly

File: test/scripts/dynamic-fee-teal-test.sh (hunk 4 above)
Comment (justicz) [reply test/scripts/dynamic-fee-teal-test.sh]: updated, thanks

File: test/scripts/dynamic-fee-teal-test.sh (hunk 5 above)
Comment (justicz) [reply test/scripts/dynamic-fee-teal-test.sh]: changed

**Pull request merged**